src/Connection.php

```diff
@@ -212,21 +213,31 @@ public function scrollSelect($params, $bindings = [])
 
         $numResults = count($results['hits']['hits']);
 
-        if ( $limit >= $numResults ){
-            $results['scrollCursor'] = $this->scroll($scrollId, $scrollTimeout, $limit - $numResults);
+        foreach ($results['hits']['hits'] as $result) {
```

Comment:
I don't really understand this bit. Why do you need to yield results in two places?

Reply:
Sorry, that part isn't very clear. The way I understand ES scroll working is that you do an initial `select()` including additional scroll params; that `select()` returns the first batch of results as well as a scroll cursor ID that you can use to make subsequent requests, each of which returns another batch and the scroll ID (or a new one) to be used for the next request.

I think it's probably possible to neaten this up so that we're not yielding twice but I haven't quite get my head around how.